backend/users/autocomplete.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+from django.contrib.admin.widgets import SELECT2_TRANSLATIONS, AutocompleteSelect
+from django.urls import reverse
+from django.utils.translation import get_language
+
+from users.client import get_users_data_by_ids
+
+
+class UsersBackendAutocomplete(AutocompleteSelect):
+    def __init__(self, admin_site, attrs=None, choices=(), using=None):
+        self.admin_site = admin_site
+        self.db = using
+        self.choices = choices
+        self.attrs = {} if attrs is None else attrs.copy()
+        self.i18n_name = SELECT2_TRANSLATIONS.get(get_language())
+
+    def build_attrs(self, base_attrs, extra_attrs=None):
+        """
+        Set select2's AJAX attributes.
+        Attributes can be set using the html5 data attribute.
+        Nested attributes require a double dash as per
+        https://select2.org/configuration/data-attributes#nested-subkey-options
+        """
+        attrs = {}
+        attrs.setdefault("class", "")
+        attrs.update(
+            {
+                "data-ajax--cache": "true",
+                "data-ajax--delay": 250,
+                "data-ajax--type": "GET",
+                "data-ajax--url": reverse("admin:users-admin-autocomplete"),
+                "data-theme": "admin-autocomplete",
+                "data-allow-clear": json.dumps(not self.is_required),
+                "data-placeholder": "",  # Allows clearing of the input.
+                "lang": self.i18n_name,
+                "class": attrs["class"]
+                + (" " if attrs["class"] else "")
+                + "admin-autocomplete",
+            }
+        )
+        return attrs
+
+    def optgroups(self, name, value, attr=None):
+        default = (None, [], 0)
+        groups = [default]
+
+        if value and value[0]:
+            users_by_id = get_users_data_by_ids(value)
```

Comment:
do you think we can find a way to move this http call somewhere more visible?
I find it a bit worrying that it is hidden in this `optgroups` function

Reply:
`optgroups` is called the first time to prepare the UI to show what is already set

Not 100% sure what you find worrying about it behind inside optgroups? How would you make it more visibile?

Reply:
as far as I know we can't call this somewhere else, since we only have access to the initial form state here as far as I know

Reply:
I'd do the call outside and pass the data the autocomplete :) but it's not a big deal, as this is for the django admin

Reply:
uhm but we only have the data in here?